octobot_trading/producers/ohlcv_updater.py

```diff
@@ -31,27 +31,42 @@ class OHLCVUpdater(OHLCVProducer):
     OHLCV_ON_ERROR_TIME = 5
     OHLCV_MIN_REFRESH_TIME = 3
 
+    OHLCV_INITIALIZATION_TIMEOUT = 60
+
     def __init__(self, channel):
         super().__init__(channel)
         self.tasks = []
         self.is_initialized = False
 
+        self.ohlcv_initialized_event = asyncio.Event()
+
+    async def wait_for_initialization(self):
+        await asyncio.wait_for(self.ohlcv_initialized_event.wait(), self.OHLCV_INITIALIZATION_TIMEOUT)
```

Comment:
Do you think we can accept to update candle even if the initialization process timeout ?

Reply:
If it timeout, it means we have no historical ohlcv, or it's extremely slow to fetch. I think in this case it's better to keep going with backtesting, if there are more problems with ohlcv fetching, we will now it soon enough. I don't think it's work cancelling the whole backtesting though: if subsequent calls work then backtesting should keep going

Reply:
But here is ohlcv_updater, it's not only a backtesting updater. You should not think backtesting only.

Reply:
For now this function is only used in backtesting (not necessary for live data), however in case we can to know if ohlcv are initialized in live data, it might make a difference, i'll add an optionnal timeout param.

Reply:
Ok but if this method only concerns backtesting, why it has to be implemented in `ohlcv_updater.py` and not just in `ohlcv_updater_simulator.py` ?

Reply:
It's called by ```recent_trade_updater_simulator``` as a return value of ```ohlcv_chan.get_producers()```, we technically car put it in ```ohlcv_updater_simulator.py``` but if we use ```recent_trade_updater_simulator``` with another updater than ```ohlcv_updater_simulator.py```, it will crash. We can consider it acceptable though, what do you think ?

Reply:
I understand, maybe the cleaner way is to have an abstract updater simulatr with an abstract wait_for_initialization ? That prevents a simulator to not have wait_for_initialization method. What do you think ?

Reply:
I think abstracting it is a good idea, we can make ```wait_for_initialization``` raise an NotImplemented exception on ohlcv_updater.py and have it implemented on ohlcv_updater_simulator.py. This way if we want to implement it later we can and it's explicit. I think it's better than creating a new abstract class that will result in multiple inheritence and won't be cythonizable

Reply:
:+1: